src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc

```diff
@@ -18,78 +18,15 @@ summary:
 
 :context: defining-custom-commands-for-che-theia
 
-The Che-Theia IDE allows users to define custom commands in a devfile that are then available when working in a workspace.
+When starting a new project, new developers often loose time in finding the good procedure to build, run or debug a project. The result is documentation that are hard to follow and never up-to-date.
```

Comment:
Hi @sunix, thanks for these updates. I believe these first two substitutions are using language that is better suited for a blogpost, fox example. The original wording was more in line with what is expected of docs.

Reply:
Maybe, the goal is to give more context so users would understand when and why to use custom commands.

Reply:
Previous wording was not helping in that purpose.

Reply:
@rkratky is this fine for you?

Reply:
To me it reads more like a marketing pitch than documentation.

Reply:
if a marketing pitch helps the user to better understand why and when (the context) to use custom commands ... it is very fine :)

Reply:
I disagree, when I look for documentation on a program I'm using, if I see ad copy I'm likely to never look at that documentation again. The documentation should answer the question "How do I add custom commands to my workspace?" -- if I'm asking the question I don't need some sentences justifying why it's a good idea.

Reply:
This doc should answer the questions:
1. what is a che commands. including in which context a user should use it.
2. how do I add commands in my devfiles/workspaces

If it feels like marketing pitch for you, feel free to rephrase it. But it should include 1.

Reply:
@sunix I still disagree. We need to take into account that the majority of users who read documentation do not read all of it. They do not open it on page one and read it like a book until the end. They want to find a quick solution to their problem, or want to follow simple instructions on how to perform specific tasks.

The text that you're suggesting assumes that there would be readers who would want to read the documentation because of their interest in the application. You're describing the reasoning behind Che philosophy and offering insight into why certain features might be of interest to someone. That's important, but it's not the function of documentation. This would be for a product website, blogpost, presentation, or a magazine article.

Let's keep the documentation focused on specific tasks.

Reply:
@rkratky
> there would be readers who would want to read the documentation because of their interest in the application. 
Of course we have people reading the doc and read the doc because of their interest in the app!
They come to the documentation because they don't know the project and how to use it. With the current state of the doc, these users are just giving up because the docs is not helping them enough to understand and make good devfiles for their dev team.

Here we are describing how to use a feature but if we don't explain what is this feature about or for, it is useless.
This is not a troubleshooting section, but a user guide. These users don't know Che and what is the difference between VSCode and Che. To me this is very important.
OK this part could be somewhere else but it is important to link the custom commands to the Che "philosophy" and why custom commands in devfile would help to developers in the daily coding life.

BTW, this is just 3 lines as an intro/context, people could skip it and go to the right section if they want.

Reply:
> We need to take into account that the majority of users who read documentation do not read all of it.

That's why recalling the context is important in this single page.